### PR TITLE
Massive Action: link multiple tickets to a problem

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Internal ref: 18423, 19662.

New massive action to link multiple tickets to a problem.
image

New massive action to resolve multiple tickets.
image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/ticket.class.php [hunk 1]
```diff
@@ -2565,6 +2571,14 @@ static function showMassiveActionsSubForm(MassiveAction $ma) {
             ]);
             echo "</td></tr></table>";
             return true;
+
+         case 'link_to_problem':
+            Problem::dropdown(['name' => 'problems_id']);
+            echo '<br><br>';
+            echo Html::submit(_('Link'), [
```
Comment (trasher) [inline inc/ticket.class.php:2578]:
```suggestion
            echo Html::submit(__('Link'), [
```

**Comment by AdrienClairembault** [comment]
Are you still expecting changes or is this good to merge ?

File: inc/ticket.class.php [hunk 2]
```diff
@@ -2565,6 +2579,73 @@ static function showMassiveActionsSubForm(MassiveAction $ma) {
...
+               'name'      => 'link'
+            ]);
+            return true;
+
+         case 'resolve_tickets':
+            $rand_type = mt_rand();
+            echo '<div class="horizontal-form">';
+
+            echo '<div class="form-row">';
+            $label = _n('Solution template', 'Solution templates', 1);
```
Comment (trasher) [inline inc/ticket.class.php:2596]:
```suggestion
            $label = SolutionTemplate::getTypeName(1);
```

File: inc/ticket.class.php [hunk 3]
```diff
@@ -2565,6 +2579,73 @@ static function showMassiveActionsSubForm(MassiveAction $ma) {
...
+                  'url'             => $CFG_GLPI["root_doc"]. "/ajax/solution.php",
+                  'moreparams' => [
+                     'type_id' => "dropdown_solutiontypes_id$rand_type"
+                  ]
+               ]
+            ]);
+            echo '</div>'; // .form-row
+
+            echo '<div class="form-row">';
+            $label = __('Solution type');
```
Comment (trasher) [inline inc/ticket.class.php:2614]: Will cause gettext warnings.
```suggestion
            $label = SolutionType::getTypeName(1);
```

**Review by trasher — CHANGES_REQUESTED** [review]

**Comment by trasher** [comment]
Please rebase on up to date master branch, to get new tests (especially those on gettext).

**Comment by AdrienClairembault** [comment]
Rebased again to fix conflict, should be good to merge.

**Review by trasher — APPROVED** [review]

**Pull request merged**